### PR TITLE
Potential fix for failing tests on travis

### PR DESCRIPTION
**Pull request by jrbenny35**
Trying to fix these failing tests on nightly. I think it is a timeout issue but I can't reproduce the failures locally.

### CODE REVIEW

**Comment by jrbenny35** [comment]
@davehunt any idea what this could be? I've updated geckodriver and tried extending the timeout.

File: .travis.yml [hunk 1]
```diff
@@ -75,9 +75,10 @@ matrix:
     - python: 2.7
     - python: 3.6
 before_install:
-  - wget -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.13.0/geckodriver-v0.13.0-linux64.tar.gz
-  - mkdir $HOME/geckodriver && tar xvf /tmp/geckodriver.tar.gz -C $HOME/geckodriver
-  - export PATH=$HOME/geckodriver:$PATH
+  - export GECKODRIVER_DOWNLOAD=`curl -s 'https://api.github.com/repos/mozilla/geckodriver/releases/latest' | python -c "import sys, json; r = json.load(sys.stdin); print([a for a in r['assets'] if 'linux64' in a['name']][0]['browser_download_url']);"`
+  - curl -L -o geckodriver.tar.gz $GECKODRIVER_DOWNLOAD
+  - gunzip -c geckodriver.tar.gz | tar xopf -
+  - chmod +x geckodriver && sudo mv geckodriver /usr/local/bin
```
Comment (davehunt) [inline .travis.yml]: We should avoid introducing sudo.. let's just extract to a subdirectory and add that to the path, like we did previously.

**Comment by davehunt** [comment]
I'm able to replicate the failures locally - are you not? I suspect there have been changes to the notifications covered by these tests that we'll need to support in Nightly builds.

**Comment by davehunt** [comment]
So there's a new popup notification on Nightly with id `addon-webext-permissions-notification`. I was able to get most of the tests passing, but ran out of time before a meeting to fix one that needs to access the add-on name. We'll also need to make sure we have some logic in FoxPuppet to conditionally expect different popups depending on the version of Firefox.

**Comment by davehunt** [comment]
Here's my work in progress, in case you want to pick this up:

```patch
diff --git a/foxpuppet/windows/browser/notifications/addons.py b/foxpuppet/windows/browser/notifications/addons.py
index dda2b56..039f558 100644
--- a/foxpuppet/windows/browser/notifications/addons.py
+++ b/foxpuppet/windows/browser/notifications/addons.py
@@ -78,6 +78,33 @@ class AddOnProgress(BaseNotification):
     """Add-on progress notification."""
 
 
+class AddOnWebExtPermissions(BaseNotification):
+    """Add-on web extension permissions notification."""
+
+    _addon_name_locator = (By.CSS_SELECTOR, '.addon-webext-name')
+
+    @property
+    def addon_name(self):
+        """Provide access to the add-on name.
+
+        :returns: The add-on name.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            return self.root.find_element(*self._addon_name_locator).text
+
+    def add(self):
+        """Confirm add-on install."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.root.find_anonymous_element_by_attribute(
+                'anonid', 'button').click()
+
+    def cancel(self):
+        """Cancel add-on install."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.root.find_anonymous_element_by_attribute(
+                'anonid', 'secondarybutton').click()
+
+
 NOTIFICATIONS = {
     'addon-install-blocked-notification': AddOnInstallBlocked,
     'addon-install-confirmation-notification': AddOnInstallConfirmation,
@@ -85,4 +112,5 @@ NOTIFICATIONS = {
     'addon-install-restart-notification': AddOnInstallRestart,
     'addon-install-failed-notification': AddOnInstallFailed,
     'addon-progress-notification': AddOnProgress,
+    'addon-webext-permissions-notification': AddOnWebExtPermissions,
 }
diff --git a/tests/test_notifications.py b/tests/test_notifications.py
index 5ea7c1a..9f93067 100644
--- a/tests/test_notifications.py
+++ b/tests/test_notifications.py
@@ -9,7 +9,7 @@ from foxpuppet.windows.browser.notifications import BaseNotification
 from foxpuppet.windows.browser.notifications.addons import (
     AddOnInstallBlocked,
     AddOnInstallComplete,
-    AddOnInstallConfirmation)
+    AddOnWebExtPermissions)
 
 
 @pytest.fixture
@@ -39,7 +39,7 @@ def blocked_notification(addon, browser, webserver, selenium):
 @pytest.fixture
 def confirmation_notification(browser, blocked_notification):
     blocked_notification.allow()
-    return browser.wait_for_notification(AddOnInstallConfirmation)
+    return browser.wait_for_notification(AddOnWebExtPermissions)
 
 
 def test_open_close_notification(browser, blocked_notification):
@@ -75,7 +75,7 @@ def test_notification_with_origin(browser, webserver, blocked_notification):
 def test_allow_blocked_addon(browser, blocked_notification):
     """Allow a blocked add-on installation"""
     blocked_notification.allow()
-    browser.wait_for_notification(AddOnInstallConfirmation)
+    browser.wait_for_notification(AddOnWebExtPermissions)
 
 
 def test_cancel_addon_install(browser, confirmation_notification):
```

**Comment by jrbenny35** [comment]
I applied your changes but it seems they fail on my version on firefox. My nightly is updated too. I wonder what exactly triggers the difference. Is it just the version? Maybe >55 ?

**Comment by jrbenny35** [comment]
What needs to be done for the name?

**Comment by davehunt** [comment]
> What needs to be done for the name?

My patch is incomplete and I had to stop for a meeting. We just need to determine the correct locator for the addon name in Firefox 55. Use the inspector in the Browser Toolbox, which should help you discover the appropriate locator.

I would also suggest removing `AddOnWebExtPermissions` from my patch and make `AddOnInstallConfirmation` conditional based on the version of Firefox. That should make it a little easier to manage.

**Comment by jrbenny35** [comment]
@davehunt did you see any changes with the install conformation notification?

**Comment by davehunt** [comment]
> did you see any changes with the install conformation notification?

Yes, this is the notification that has been replaced. It might be that the new one is exclusive for web extensions, but as legacy add-ons are going away it's probably safe to reuse this class for the new notification as the name is still descriptive and the actions are identical.

**Comment by jrbenny35** [comment]
Well it seems after clicking install the notification shown is different as the test keep failing no matter what notification I try and wait for.```selenium.common.exceptions.TimeoutException: Message: AddOnWebExtPermissions was not shown.```

**Comment by davehunt** [comment]
@jrbenny35 strange, it was working for me. I'll look again tomorrow.

**Comment by davehunt** [comment]
Updated since we've fixed the failures. I still think we should merge this once the comments are addressed though, so we're always using latest geckodriver.

**Comment by jrbenny35** [comment]
Well I could close this and open a new one just for geckodriver.

**Comment by davehunt** [comment]
> Well I could close this and open a new one just for geckodriver.

Pretty much all this patch currently does is update geckodriver, but it's up to you if you want to do it in another pull request.

**Comment by davehunt** [comment]
Let's close this in favour of #86

**Pull request closed without merging**